com.aobuchow.themes.spectrum.preferences/src/com/aobuchow/themes/spectrum/preferences/ColorHSL.java

```diff
@@ -0,0 +1,121 @@
+package com.aobuchow.themes.spectrum.preferences;
+
+import org.eclipse.swt.graphics.Color;
+import org.eclipse.swt.graphics.RGB;
+import org.eclipse.ui.PlatformUI;
+
+public class ColorHSL {
+	private Color color;
+	private float hue;
+	private float saturation;
+	private float luminance;
+
+	public ColorHSL(Color color) {
+		this.color = color;
+		this.hue = color.getRGB().getHSB()[0];
+		this.saturation = color.getRGB().getHSB()[1];
+		this.luminance = color.getRGB().getHSB()[2];
+	}
+
+	public ColorHSL(float hue, float saturation, float luminance) {
+		this(new Color(PlatformUI.getWorkbench().getDisplay(), new RGB(hue, saturation, luminance)));
+	}
+
+	//TODO: Document the enums
+	public ColorHSL modifyProperty(MODIFICATION modificationType, HSL_PROPERTY propertyToModify, BOUND_BEHAVIOR boundBehavior, float amount) {
```

Comment:
What if you only want to modify HUE and not modify SATURATION or LUMINANCE? Maybe we could add some KEEP value to all of the enum?

Another thought: In terms of OpenClosed principle we maybe could decouple modification from the actual data storage. On the other hand, maybe this approach right now keeps things simpler (considering that this will never be API of the bundle.

Reply:
Suggestion: We could add some test fragment w/ tests for this class. :) (@AObuchow since you already mentioned this, I thought it cannot hurt to add this note :)

Reply:
> What if you only want to modify HUE and not modify SATURATION or LUMINANCE? Maybe we could add some KEEP value to all of the enum?

You're able to modify only one of the HSL if desired, see [here](https://github.com/AObuchow/Eclipse-Spectrum-Theme/pull/114/files#diff-aa9864b8580ec6b48777306890e8a78dR91) in ColorManager where it's being used. Is that what you meant?
> 
> Another thought: In terms of OpenClosed principle we maybe could decouple modification from the actual data storage. On the other hand, maybe this approach right now keeps things simpler (considering that this will never be API of the bundle.

I do agree with this, but I also have another opinion: the storage of the HSL is really in the RGB class currently, so providing a class that only acts a wrapper to get HSL properties seemed a little bit.. less exciting ;) I also wanted to use the builder pattern, and it seemed natural to be able to build a new color from an existing color by applying HSL transformations.

However... this builder class could also be in another class such as ColorUtils. 

Maybe I'll keep things this way for this PR, but I'm very open to moving things around to make the API nicer. I'd like to upstream this API, but if that isn't possible, I'd be willing to make this API usable from other bundles.

Reply:
> Suggestion: We could add some test fragment w/ tests for this class. :) (@AObuchow since you already mentioned this, I thought it cannot hurt to add this note :)

+1 for adding tests for this class in particular. I might make another bug for this and work on it after this PR is merged.

Reply:
> > What if you only want to modify HUE and not modify SATURATION or LUMINANCE? Maybe we could add some KEEP value to all of the enum?
> 
> You're able to modify only one of the HSL if desired, see [here](https://github.com/AObuchow/Eclipse-Spectrum-Theme/pull/114/files#diff-aa9864b8580ec6b48777306890e8a78dR91) in ColorManager where it's being used. Is that what you meant?
> 
> > Another thought: In terms of OpenClosed principle we maybe could decouple modification from the actual data storage. On the other hand, maybe this approach right now keeps things simpler (considering that this will never be API of the bundle.
> 
> I do agree with this, but I also have another opinion: the storage of the HSL is really in the RGB class currently, so providing a class that only acts a wrapper to get HSL properties seemed a little bit.. less exciting ;) I also wanted to use the builder pattern, and it seemed natural to be able to build a new color from an existing color by applying HSL transformations.
> 
> However... this builder class could also be in another class such as ColorUtils.
> 
> Maybe I'll keep things this way for this PR, but I'm very open to moving things around to make the API nicer. I'd like to upstream this API, but if that isn't possible, I'd be willing to make this API usable from other bundles.

I think I misread the modify-API. Maybe I should have given it a closer look. :) I understood it as a way to modify a color in 3 dimensions - but it's not that way. So, ignore that comment. Sorry ;)

Regarding your Storage-vs-modifcation opinion: Perfectly relatable :)

Reply:
No worries, it could be documented better :P And okay glad you see my perspective ;)